assets/js/main.js

```diff
@@ -0,0 +1,11 @@
+import { TimeTree } from "./timetree";
+
+// load & parse leaf data from json embedded in the document
+const data = JSON.parse(document.querySelector(".leaf-data").value);
+// load and parse tag list for label / slug lookup label based on slug
+const tags = JSON.parse(document.querySelector(".tag-data").value);
+// determine if this is a production or development build
+const params = JSON.parse(document.querySelector(".env-data").value);
```

Comment:
More of a question - interested why you chose to inject the data this way rather than through an embedded script or reading a CSV / JSON blob with D3. Had a similar problem come up recently and went back and forth on different approaches. The way it's rendered isn't very readable so that doesn't seem to be an advantage.

Reply:
Thanks for flagging; I don't remember why I did it this way! I agree that using a script tag with `type="application/json"` is better. Just figured out how to generate that with Hugo (had to figure out how to tell it not to escape my json 😀 ) and updated the parse calls.